### PR TITLE
Android gdbserver

### PR DESCRIPTION
**Pull request by akiannillo**
Gdbserver based on tcp:PORT instead of files in order to properly work with gdb and target remote.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -70,14 +70,14 @@ if ARCH_ARM64
 endif
 
 check: push
-	adb shell "\"${RPFX}/tests/gum-tests\""
+	adb shell "\"${RPFX}/tests/gum-tests\" -p /Core/Arm64Stalker"
```
Comment (oleavr) [inline Makefile.am]: Unrelated change.

File: Makefile.am [hunk 2]
```diff
@@ -70,14 +70,14 @@ if ARCH_ARM64
...
 
 check-gdb: $(top_builddir)/tests/android-gdb.setup
 	adb shell "${RM} -f /data/local/tmp/tests/debug-socket"
 	adb push "${ANDROID_NDK_ROOT}/prebuilt/android-$(android_arch)/gdbserver/gdbserver" "${RPFX}/tests/"
-	adb shell "${RPFX}/tests/gdbserver +${RPFX}/tests/debug-socket ${RPFX}/tests/gum-tests" &
+	adb shell "${RPFX}/tests/gdbserver :${DEBUG_PORT} ${RPFX}/tests/gum-tests -p /Core/Arm64Stalker/performance" &
```
Comment (oleavr) [inline Makefile.am]: The `-p` part here is unrelated.

Comment (akiannillo) [inline Makefile.am]: OMG, sorry for that. My bad.
